core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -867,20 +848,27 @@ public boolean setForcedVariation(@Nonnull String experimentKey,
     @Nullable
     public Variation getForcedVariation(@Nonnull String experimentKey,
                                         @Nonnull String userId) {
-        if (!isValid) {
+        ProjectConfig projectConfig = getProjectConfig();
+        if (projectConfig == null) {
             logger.error("Optimizely instance is not valid, failing getForcedVariation call.");
             return null;
         }
 
-        return projectConfig.getForcedVariation(experimentKey, userId);
+        Experiment experiment = projectConfig.getExperimentKeyMapping().get(experimentKey);
+        if (experiment == null) {
+            logger.debug("No experiment \"{}\" mapped to user \"{}\" in the forced variation map ", experimentKey, userId);
+            return null;
+        }
+
+        return decisionService.getForcedVariation(experiment, userId);
     }
 
     /**
      * @return the current {@link ProjectConfig} instance.
      */
-    @Nonnull
+    @Nullable
```

Comment:
I think this represents a breaking change right? This will cause compilation failures when the user updates to the latest 3.x.x version

Reply:
It's not a compilation failure, but more of a hint to your IDE and static code analysis since it can't be fully respected at runtime. The reason I changed it is because it is nullable today even before this change :) Marking as `@Nonnull` can lead to uncaught NPEs.

For example trace [Optimizely#initialize](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/Optimizely.java#L114) to see how this can be null better yet check out this [test](https://github.com/optimizely/java-sdk/blob/master/core-api/src/test/java/com/optimizely/ab/OptimizelyTest.java#L1567) to see what would be returned.